### PR TITLE
Feature IM-11 Automatic Testing with JUnit

### PR DESCRIPTION
**Pull request by tilokowalski**

### CODE REVIEW

File: src/test/java/com/fes/hazelnut/schoolnut/data/entity/enrollment/EnrollmentServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import static org.mockito.Mockito.when;
+
+import java.util.Optional;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.fes.hazelnut.schoolnut.data.entity.schoolclass.SchoolclassEntity;
```
Comment (tilokowalski) [inline src/test/java/com/fes/hazelnut/schoolnut/data/entity/enrollment/EnrollmentServiceTest.java:15]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Extra separation in import group before 'com.fes.hazelnut.schoolnut.data.entity.schoolclass.SchoolclassEntity'

Comment (tilokowalski) [inline src/test/java/com/fes/hazelnut/schoolnut/data/entity/enrollment/EnrollmentServiceTest.java:15]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'com.fes.hazelnut.schoolnut.data.entity.schoolclass.SchoolclassEntity' import. Should be before 'org.mockito.junit.jupiter.MockitoExtension'.

File: src/test/java/com/fes/hazelnut/schoolnut/data/entity/enrollment/EnrollmentServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    @Test
+    public void findOneByCode_whenEnrollmentExists_thenExistingEnrollmentShouldBeReturned() {
+        EnrollmentEntity expected = new EnrollmentEntity();
+        when(repository.findOneByCode("code")).thenReturn(Optional.of(expected));
+        
+        Optional<EnrollmentEntity> actual = service.findOneByCode("code");
+        assertThat(actual).isEqualTo(Optional.of(expected));
+    }
+
+    @Test
```
Comment (tilokowalski) [inline src/test/java/com/fes/hazelnut/schoolnut/data/entity/enrollment/EnrollmentServiceTest.java:60]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.MissingJavadocMethodCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing a Javadoc comment.

File: src/test/java/com/fes/hazelnut/schoolnut/data/entity/enrollment/EnrollmentServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    public void findOneByCode_whenEnrollmentDoesNotExist_thenEmptyOptionalShouldBeReturned() {
+        when(repository.findOneByCode("code")).thenReturn(Optional.empty());
+
+        Optional<EnrollmentEntity> actual = service.findOneByCode("code");
+        assertThat(actual).isEqualTo(Optional.empty());
+    }
+
+    @Test
```
Comment (tilokowalski) [inline src/test/java/com/fes/hazelnut/schoolnut/data/entity/enrollment/EnrollmentServiceTest.java:68]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.MissingJavadocMethodCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing a Javadoc comment.

File: src/test/java/com/fes/hazelnut/schoolnut/data/entity/enrollment/EnrollmentServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    public void deleteAll() {
+        service.deleteAll();
+
+        verify(repository, times(1)).deleteAllInBatch();
+    }
+
+    @Test
```
Comment (tilokowalski) [inline src/test/java/com/fes/hazelnut/schoolnut/data/entity/enrollment/EnrollmentServiceTest.java:75]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.MissingJavadocMethodCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing a Javadoc comment.

File: src/main/java/com/fes/hazelnut/schoolnut/data/entity/enrollment/EnrollmentService.java [hunk 5]
```diff
@@ -10,6 +10,18 @@
  */
 public interface EnrollmentService {
 
+    /**
+     * Saves an erollment.
```
Comment (lenav99) [inline src/main/java/com/fes/hazelnut/schoolnut/data/entity/enrollment/EnrollmentService.java:14]: enrollment!

**Review by VogtJoshua — APPROVED** [review]

File: src/test/java/com/fes/hazelnut/schoolnut/data/entity/enrollment/EnrollmentServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+        EnrollmentEntity expected = new EnrollmentEntity();
+        when(repository.findOneByCode(CODE)).thenReturn(Optional.of(expected));
+        
+        Optional<EnrollmentEntity> actual = service.findOneByCode(CODE);
+        assertThat(actual).isEqualTo(Optional.of(expected));
+    }
+
+    /**
+     * Tests {@link EnrollmentService#findOneByCode(String)}.
+     * Verifies that the repository's findOneByCode returns an empty optional, when the enrollment does not exist.
```
Comment (lenav99) [inline src/test/java/com/fes/hazelnut/schoolnut/data/entity/enrollment/EnrollmentServiceTest.java:95]: , when stimmt so auch nicht, ohne komma

File: src/test/java/com/fes/hazelnut/schoolnut/data/entity/schoolclass/SchoolclassServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+        List<SchoolclassEntity> expected = Arrays.asList(new SchoolclassEntity(), new SchoolclassEntity());
+        when(repository.findAll()).thenReturn(expected);
+
+        List<SchoolclassEntity> actual = service.findAll();
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    /**
+     * Tests {@link SchoolclassService#findOneByTitle(String)}.
+     * Verifies that the repository's findOneByTitle returns the expected schoolclass entity, when it exists.
```
Comment (lenav99) [inline src/test/java/com/fes/hazelnut/schoolnut/data/entity/schoolclass/SchoolclassServiceTest.java:84]: , when

File: src/test/java/com/fes/hazelnut/schoolnut/data/entity/schoolclass/SchoolclassServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+        List<SchoolclassEntity> expected = Arrays.asList(new SchoolclassEntity(), new SchoolclassEntity());
+        when(repository.findByTitleContains(FILTER)).thenReturn(expected);
+
+        List<SchoolclassEntity> actual = service.findByTitleContains(FILTER);
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    /**
+     * Tests {@link SchoolclassService#findByTitleAndPassword(String, String)}.
+     * Verifies that the repository's findByTitleAndPassword returns the expected schoolclass entity, when it exists.
```
Comment (lenav99) [inline src/test/java/com/fes/hazelnut/schoolnut/data/entity/schoolclass/SchoolclassServiceTest.java:110]: , when

File: src/test/java/com/fes/hazelnut/schoolnut/data/entity/specialization/SpecializationServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+        List<SpecializationEntity> expected = Arrays.asList(new SpecializationEntity(), new SpecializationEntity());
+        when(repository.findAll()).thenReturn(expected);
+
+        List<SpecializationEntity> actual = service.findAll();
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    /**
+     * Tests {@link SpecializationService#findOneById(long)}.
+     * Verifies that the repository's findOneById returns the expected specialization entity, when it exists.
```
Comment (lenav99) [inline src/test/java/com/fes/hazelnut/schoolnut/data/entity/specialization/SpecializationServiceTest.java:48]: , when

File: src/test/java/com/fes/hazelnut/schoolnut/security/UserDetailsServiceTest.java [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+
+/**
+ * Unit tests for {@link UserDetailsServiceImpl}.
+ * Tests the functionality of loading user details by username.
+ */
+@ExtendWith(MockitoExtension.class)
+class UserDetailsServiceTest {
+
+    /**
+     * Knowsn username to be used in the tests.
```
Comment (lenav99) [inline src/test/java/com/fes/hazelnut/schoolnut/security/UserDetailsServiceTest.java:29]: Known

**Review by lenav99 — DISMISSED** [review]
Passt so, aber Grammatik und Typos noch fixen bitte

File: src/test/java/com/fes/hazelnut/schoolnut/data/entity/config/ConfigServiceTest.java [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+    public void testSave() {
+        ConfigEntity config = new ConfigEntity();
+        service.save(config);
+
+        verify(repository, times(1)).saveAndFlush(config);
+    }
+
+    /**
+     * Tests {@link ConfigService#getConfig()}.
+     * Verifies that the repository's findOneById returns the single expected config entity, when it exists.
```
Comment (thisami) [inline src/test/java/com/fes/hazelnut/schoolnut/data/entity/config/ConfigServiceTest.java:47]: Zitat Lena zu dem Komme vor when: "Ja ne, ist nie richtig eigentlich"

**Review by thisami — DISMISSED** [review]

**Review by thisami — APPROVED** [review]

**Review by lenav99 — APPROVED** [review]

**Review by lenav99 — APPROVED** [review]

**Review by thisami — APPROVED** [review]

**Review by VogtJoshua — APPROVED** [review]

**Pull request merged**